//package core.io.protocol;

import "BattleInfo.proto";
import "BattleAction.proto";
import "BattleSettle.proto";
message DseBattleOpt
{
	optional int32			opt_type 		= 1;//1准备好 2播放action结束 3释放滑动技能 4释放怒气技能
	optional string 		id 				= 2;//技能释放式神ID
	optional int32			param			= 3;//其他参数 比如怒气技能的打P效果 1 2 3 4 
	optional int32			res				= 4;//操作返回码 0代表成功直接展示action 非0代表非法请求
	repeated BattleAction 	action 			= 5;//攻击数据
	optional bool			isover 			= 6;//是否战斗结束 isover为false时 battleinfo==null
	optional BattleInfo		battleinfo 		= 7;//整个战斗过程和初始阵容信息
	optional bool			sendclient		= 8;//是否发送回客户端
	optional BattleSettle   BattleSettle    = 9;//战斗结算信息
	optional int32			transparam		= 10;//无逻辑 转发客户端参数
	optional bool			satifycon		= 11;//封魔塔挑战是否满足条件
}


